Support allow_vendor_change setting in dnf context API

Conan-Kudo opened the pull request.

j-mracek commented:
I suggest that it will not work correctly. You can set it in global config, it is read from config, but no one set it to sack from where goal takes it.

Conan-Kudo commented:
@j-mracek I think I've fixed it now. Can you take a look?

j-mracek commented on libdnf/dnf-context.cpp:
```diff
@@ -1409,6 +1426,22 @@ dnf_context_set_install_weak_deps(gboolean enabled)
+ * Since: 0.55.2
+ */
+void
+dnf_context_set_allow_vendor_change(DnfContext *context, gboolean vendorchange)
```
It will work with microdnf, but it is redundant. See https://github.com/rpm-software-management/microdnf/pull/88/files#r526049471.
It will not work with Packagekit and may be even for Microdnf (sack is not available).
I suggest that you can return to the original code where you modified main conf. Then in `dnf_context_setup_sack_with_flags` you can set vendor change value according to value in main conf.
What do you think?
PS: Any time when you use sack from context you have to consider that it could be NULL. Specially with packagekit this is often the right suggestion, because it handles sacks, goals outside of the context.

Conan-Kudo replied:
That makes sense to me. Let me roll back to that and try that.

Conan-Kudo committed:
Support allow_vendor_change setting in dnf context API

= changelog =
msg: Support allow_vendor_change setting in dnf context API
type: enhancement

Conan-Kudo commented:
@j-mracek I've reverted back to my original approach and added configuration into `dnf_context_setup_sack_with_flags()`. Could you please take a look?

j-mracek commented:
LGTM

j-mracek commented:
@rh-atomic-bot r+

Conan-Kudo commented:
@rh-atomic-bot retry

Conan-Kudo commented:
@rh-atomic-bot retry

lukash commented:
@rh-atomic-bot retry

lukash commented:
I've run 'retry' on a wrong PR, cancelled the job.

In addition to instability, the CI is broken, this needs to get merged first: https://github.com/rpm-software-management/dnf-plugins-extras/pull/187

Sorry about the inconvenience.

j-mracek commented:
@rh-atomic-bot retry

Conan-Kudo commented:
@rh-atomic-bot retry

j-mracek commented:
Only system-upgrade tests failed, therefore not a problem with this PR.
